### PR TITLE
feat: implement telegram bot plugin

### PR DESCRIPTION
**Pull request by athurg**
This PR add Telegram Bot support inside the Memos. Core features include:

- Support both text and photo message, include multiple photo message.
- One Telegram Bot need for whole Memos Server, every user can bind there Telegram UserID in UserSettings of Memos .
- Pull mode of Telegram Bot, which works on your office LAN or home NAS.

Because it's a large PRs, so some features will be added in next version, such as:
- Webhook mode of Telegram Bot, which for people who cannot touch Telegram API.
- Support message other than text and photos, such as File Message, Location Message, etc.

### CODE REVIEW

File: plugin/telegram/api_edit_message.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package telegram
+
+import (
+	"context"
+	"net/url"
+	"strconv"
+)
+
+// EditMessage make an editMessageText api request.
+func (r *Robot) EditMessage(ctx context.Context, chatID, messageId int, text string) (*Message, error) {
```
Comment (boojack) [inline plugin/telegram/api_edit_message.go:10]: messageId -> messageID

**Review by boojack — COMMENTED** [review]
LGTM overall. Please fix the linter errors

File: api/user_setting.go [hunk 2]
```diff
@@ -16,6 +16,8 @@ const (
 	UserSettingAppearanceKey UserSettingKey = "appearance"
 	// UserSettingMemoVisibilityKey is the key type for user preference memo default visibility.
 	UserSettingMemoVisibilityKey UserSettingKey = "memo-visibility"
+	// UserSettingTelegramUserid is the key type for telegram userid of memos user
+	UserSettingTelegramUseridKey UserSettingKey = "telegram-userid"
```
[the remark this answers is not in the document]
Comment (athurg) [reply api/user_setting.go]: Fixed

File: plugin/telegram/api_edit_message.go (hunk 1 above)
Comment (athurg) [reply plugin/telegram/api_edit_message.go]: Fixed

**Review by boojack — APPROVED** [review]
LGMT, let's move on!

**Comment by athurg** [comment]
I have rebase the branch into latest `main`, and fix all the golang static check failure.

**Pull request merged**